### PR TITLE
ci: use Travis cache to reduce build time

### PR DESCRIPTION
**Pull request by liubin**
This PR includes these changes:
- use Rust installed by Travis
- install x86_64-unknown-linux-musl
- install rustfmt
- use Travis cache
- delete ci/install_vc.sh

Signed-off-by: bin liu <bin@hyper.sh>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,8 +5,23 @@
...
+# by ways documented in
+# https://docs.travis-ci.com/user/caching#clearing-caches
+language: rust
+rust:
+  - 1.44.1
+  - stable
+cache:
+  cargo: true
+  directories:
+    - src/agent/target
```
Comment (jodh-intel) [inline .travis.yml:23]: Hi @liubin - can you explain how this works for PRs which change the agent code. Presumably, the cache is somehow ignored, or recreated in that scenario?

Comment (liubin) [reply .travis.yml:23]: The cached content includes:
- $TRAVIS_HOME/.cache/sccache
- $TRAVIS_HOME/.cargo/
- $TRAVIS_HOME/.rustup/
- src/agent/target

So if there are no dependencies changed, the cache will hit and reduce the time compiling the external crates.

Rust agent and  rustjails/protocols/... will  always be forced to  build and  not use caches.

**Review by devimc — APPROVED** [review]
lgtm - thanks @liubin

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin.

lgtm

**Pull request merged**